public/app/features/explore/TraceView/components/TraceTimelineViewer/VirtualizedTraceView.tsx

```diff
@@ -406,6 +407,12 @@ export class UnthemedVirtualizedTraceView extends React.Component<VirtualizedTra
       };
     }
 
+    const prevSpan = spanIndex > 0 ? trace.spans[spanIndex - 1] : null;
+
+    const start = Math.max((this.listView?.getTopVisibleIndex() || 0) - BUFFER_SIZE, 0);
```

Comment:
This is duplicated here and also for the span detail, but it seems like it isn't actually dependent on any information about the row. It is dependent only on the listView and it's position so I wonder if it would make sense to put in somewhere like onScroll handler or something like that. Also here I assume this will render only when it comes into view (or near depending on the buffer) and after it does not rerender does it? So I would assume this will get stuck with `visibleSpanIds` from the moment it was rendered and does not update as the view actually scrolls around and spans get visible or hidden, does it?